ENGAGEHF/VitalsManager/VitalsManager.swift

```diff
@@ -145,21 +164,18 @@ public class VitalsManager: Module, EnvironmentAccessible {
     
     
     private func convertToHKQuantitySample(_ observation: R4Observation) throws -> HKQuantitySample {
-        guard let observationType = observation.code.coding?.first?.code?.value?.string else {
-            throw VitalsError.invalidConversion
-        }
-        
         let hkQuantity: HKQuantity
         let quantityType: HKQuantityType
         
-        switch observationType {
-        case "29463-7": // Weight
+        if observation.code.containsCoding(code: "29463-7", system: FHIRSystem.loinc.url) {
+            // Weight
             hkQuantity = try self.getQuantity(observation: observation)
             quantityType = HKQuantityType(.bodyMass)
-        case "8867-4": // Heart Rate
+        } else if observation.code.containsCoding(code: "8867-4", system: FHIRSystem.loinc.url) {
```

Comment:
Can we maybe also put those codes in an enum? They are probably used more than once, e.g. when creating the observations in the first place